src/ecmascript/spidermonkey/document.c

```diff
@@ -450,25 +478,57 @@ document_write_do(JSContext *ctx, unsigned int argc, JS::Value *rval, int newlin
 	struct string *ret = interpreter->ret;
 	JS::CallArgs args = JS::CallArgsFromVp(argc, rval);
 
-	if (argc >= 1 && ret) {
-		int i = 0;
+	char *code;
 
-		for (; i < argc; ++i) {
-			char *code = jsval_to_string(ctx, args[i]);
+	code = stracpy("");
```

Comment:
You do not have to initialize with stracpy.
Working with char * is error prone. ELinks has struct string . See src/util/string.h
and many examples in code.

So you init_string(...)
add_to_string(), there is many other similar functions
and finally done_string(...)

Please rewrite your code using struct string.

Reply:
O.K., agreed. Fixed.